crates/pixi_core/src/lock_file/update.rs

```diff
@@ -456,14 +454,25 @@ impl<'p> LockFileDerivedData<'p> {
 
                 let platform = environment.best_platform();
                 let locked_env = self.locked_env(environment)?;
-                let packages =
-                    Self::filter_skipped_packages(locked_env.packages(platform), skipped);
+                let subset = InstallSubset::new(
+                    &filter.skip_with_deps,
+                    &filter.skip_direct,
+                    filter.target_package.as_deref(),
+                );
+                let result = subset.filter(locked_env.packages(platform));
+                let packages = result.install;
+                let ignored = result.ignore;
 
                 // Separate the packages into conda and pypi packages
                 let (conda_packages, pypi_packages) = packages
                     .into_iter()
                     .partition::<Vec<_>, _>(|p| p.as_conda().is_some());
 
+                // TODO: use the same ignore logic for PyPI
+                let (ignored_conda, _ignored_pypi) = ignored
```

Comment:
I've kept the behavior as before for PyPI. So that if you do:

```bash
pixi install --frozen --skip foobar`
```

It works like before, but we need to improve that, because without `--frozen` first this was required it might work unintuitively.